File: python/L2.Addition.Subtraction.ScalarMultiplication.py
from lib.vector import Vector

#Lesson 2

#2.1 Addition
v = Vector([8.218, -9.341])
w = Vector([-1.129, 2.111]) 
print("V + W =" + str(v.plus(w)))

#2.2 Subtraction
x = Vector([7.119, 8.215])
y = Vector([-8.223, 0.878])
print("X - Y = " + str(x.minus(y)))

#2.3 Scalar Multiplication
c = 7.41
z = Vector([1.671, -1.012, -0.318])
print("Scalar c x Vector z = " + str(z.times_scalar(c)))

File: js/L2.Addition.Subtraction.ScalarMultiplication.js
const Vector = require('./vector.js');

let v1 = new Vector( [8.218, -9.341]);
let v2 = new Vector([-1.129, 2.111])
console.log("Lesson 1 Problem 1: [8.218, -9.341] + [-1.129, 2.111] = " + v1.plus(v2).toString());

v21 = new Vector([7.119, 8.215]);
y21b = new Vector([-8.223, 0.878]);
console.log("Lesson 1 Problem 2: [7.119, 8.215] - [-8.223, 0.878] = " + v21.minus(y21b).toString());

let c = 7.41
let z = new Vector([1.671, -1.012, -0.318])
console.log("Lesson 1 Problem 3: 7.41 * [1.671, -1.012, -0.318] = " + z.times_scalar(c).toString());